File: java-api/src/main/java/de/davelee/personalman/api/ResetUserRequest.java
package de.davelee.personalman.api;

import lombok.*;

/**
 * This class is part of the PersonalMan REST API. It represents a request to reset the password of a particular user.
 * @author Dave Lee
 */
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ResetUserRequest {

    // company associated with
    private String company;

    // username who's password should be reset
    private String username;

    // new password to set for this user
    private String password;

    // The token of the user to verify that they are logged in
    private String token;

}

File: server/src/users/requests/resetuser.request.ts
import { ApiProperty } from "@nestjs/swagger";

export class ResetUserRequest {
    // company associated with
    @ApiProperty()
    private company: string;

    // username who's password should be reset
    @ApiProperty()
    private username: string;

    // new password to set for this user
    @ApiProperty()
    private password: string;

    // The token of the user to verify that they are logged in
    @ApiProperty()
    private token: string;

    getCompany(): string {
        return this.company;
    }

    getToken(): string {
        return this.token;
    }

    getUsername(): string {
        return this.username;
    }

    getPassword(): string {
        return this.password;
    }
}